bittide-instances/src/Bittide/Instances/Tests/FullMeshHwCc/IlaPlot.hs

```diff
@@ -0,0 +1,268 @@
+  , LocalTimestamp
+  , AccWindowHeight
+  , SyncPulsePeriod
+  , syncOutGenerator
+  , callistoClockControlWithIla
+  ) where
+
+import Clash.Explicit.Prelude
+import Clash.Explicit.Signal.Extra
+
+import Bittide.Arithmetic.Time (Milliseconds, PeriodToCycles)
+import Bittide.ClockControl (SpeedChange(..), DataCount, ClockControlConfig)
+import Bittide.ClockControl.Callisto
+  (CallistoResult(..), ReframingState(..), callistoClockControl)
+import Bittide.ClockControl.StabilityChecker
+import Bittide.Extra.Maybe (orNothing)
+
+import Clash.Cores.Xilinx.Xpm.Cdc.Handshake.Extra (xpmCdcMaybeLossy)
+import Clash.Cores.Xilinx.Xpm.Cdc.Gray (xpmCdcGray)
+import Clash.Cores.Xilinx.Ila (IlaConfig(..), Depth(..), ila, ilaConfig)
+
+import Data.Maybe (isJust, fromMaybe)
+
+type AccWindowHeight = 3 :: Nat
+type SyncPulsePeriod = Milliseconds 5
+type SyncPulseIndex dom = Index (PeriodToCycles dom SyncPulsePeriod)
+
+-- | A global timestamp consists of the number of synchronized pulses
+-- received and the number of cycles of the local stable clock
+-- (starting with 'syncStart').
+type GlobalTimestamp = (Unsigned 32, Unsigned 32)
+
+-- | The local timestamp counts the cycles of the dynamic clock
+-- (starting with clock control).
+type LocalTimestamp = Unsigned 32
+
+-- | Indicator for signaling a change in the stage of 'ReframingState'.
+data RfStageChange = Stable | ToDetect | ToWait | ToDone
+  deriving (Eq, Generic, BitPack, NFDataX)
+
+-- | Indicator for stating the reason of an ILA capture.
+data CaptureCondition = UntilTrigger | CalibrationDone | DataChange | TheEnd
+  deriving (Eq, Generic, NFDataX, BitPack)
+
+-- | Conglomerate of signals, as they are required by the ILA trigger
+-- and capture conditions.
+data IlaControl dom =
+  IlaControl
+    { syncStart :: Signal dom Bool
+      -- ^ Synchronized test start trigger
+    , syncEnd :: Signal dom Bool
+      -- ^ Synchronized test end trigger
+    , calibrate :: Signal dom Bool
+      -- ^ Calibration interval
+    , calibrationDone :: Signal dom Bool
+      -- ^ Calibration end trigger
+    , globalTimestamp :: Signal dom GlobalTimestamp
+      -- ^ Synchronized pulse counter
+    }
+
+-- | A single data type for covering all of the non-clock related data
+-- to be included into a capture.
+data PlotData (n :: Nat) (m :: Nat) =
+  PlotData
+    { dEBData       :: Vec n (DataCount m, Maybe Bool, Maybe Bool)
+    , dSpeedChange  :: SpeedChange
+    , dRfStageChange :: RfStageChange
+    }
+  deriving (Generic, NFDataX, BitPack)
+
+-- | Accumulates over multiple @FINC@/@FDEC@s, i.e. if both @FINC@ and
+-- @FDEC@ are requested after each other, then they cancel each other
+-- out and are not reported. Correspondingly, only @FINC@/@FDEC@s are
+-- reported that haven't canceled out before they exceed the @n@
+-- boundary. Thus, for example @FINC@, @FINC@, @FDEC@, @FDEC@, ... is
+-- not reported for @n > 2@ as the first two @FINC@s don't exceed the
+-- boundary @n@.
+accwindow ::
+  forall height dom.
```

Comment:
Could you explain the use of this function? I'm not sure why you'd wanna use this.

Reply:
see above
>   * `n` accumulated `FINC`/`FDEC`s (currently `n = 3`)
     _("accumulated" here means that if both `FINC` and `FDEC` are requested after each other, then they cancel each other out and are not reported. Only `FINC`/`FDEC`s are reported that haven't canceled out before they exceed the `n` boundary. Thus, for example `FINC`, `FINC`, `FDEC`, `FDEC`, ... is not reported for `n > 2` as the first two `FINC`s don't exceed the boundary `n`.)_

Reply:
I'm still not sure _why_ you'd use this. Is this to prevent too much data ending up in the ILA? I.e., it's a compression technique?

Reply:
Yes